keanu-python/keanu/algorithm/sampling.py

```diff
@@ -107,16 +111,21 @@ def sample(net: BayesNet,
     network_samples: JavaObject = sampling_algorithm.get_sampler().getPosteriorSamples(
         probabilistic_model.unwrap(), vertices_unwrapped, draws).drop(drop).downSample(down_sample_interval)
 
-    vertex_samples = {
+    vertex_samples: sample_types = {
         Vertex._get_python_label(vertex_unwrapped): list(
             map(Tensor._to_ndarray,
-                network_samples.get(vertex_unwrapped).asList())) for vertex_unwrapped in vertices_unwrapped
+                network_samples.get(vertex_unwrapped).asList(),
+                itertools.repeat(True, len(network_samples.get(vertex_unwrapped).asList()))))
+        for vertex_unwrapped in vertices_unwrapped
     }
 
     if plot:
         traceplot(vertex_samples, ax=ax)
 
-    return vertex_samples
+    if _all_vertices_are_scalar(sample_from_copy):
+        return vertex_samples
+    else:
+        return _create_multi_indexed_samples(vertex_samples, False)
```

Comment:
Could we just generate the correct type all the time?  IE we spend time building the map above - couldn't we just make it the right type first time?

Reply:
Very good point - will change